kernel/section.ml

```diff
@@ -15,11 +15,11 @@ open Declarations
 
 module NamedDecl = Context.Named.Declaration
 
-type section_entry =
-| SecDefinition of Constant.t
-| SecInductive of MutInd.t
 
 type 'a entry_map = 'a Cmap.t * 'a Mindmap.t
+type section_entry =
+| SecDefinition of Constant.t * constant_body
+| SecInductive of MutInd.t * mutual_inductive_body
```

Comment:
We need the `hyps` of the declaration in `Section.segment_entry` but not the full bodies. We could indeed use an access to the env instead. I can change it if you insist.

Reply:
Please change

Reply:
@skyskimmer: I'm still a bit afraid of the extra time needed to find again the `constant_body` and `mutual_inductive_body` in the environment. Probably not more than a dozen of steps per definition for an environment of several thousands of declarations, so it is maybe a bit unjustified fear, especially considering how lax we are on factorizing the accesses to the environment in many parts of the code, but would you be ok if I would cache back the bodies in the section_entry?

Reply:
I am only OK if you can show an impact in the bench.

Reply:
I don't think I could. That would require a huge environment and short definitions so that the time for looking in the environment dominates.

Well, ok, this is globally non-significant, so I leave the PR without the "cache".